src/day8/re_frame/trace.cljs

```diff
@@ -160,11 +161,11 @@
                                      :re-frame.router/fsm-trigger "#fd701e"
                                      nil)}}
                [:td {:style (merge row-style {:cursor "pointer"})
-                     :on-click #(swap! trace-details assoc keyword-id (not (keyword-id @trace-details)))}
-                (if (or (keyword-id @trace-details)
-                        (:showing @trace-details))
-                  "▼"
-                  "▶")]
+                     :on-click (fn [e]
+                                 (when show?
```

Comment:
This code is a little bit too clever, I think it would be better to either use an `if` to branch off `show?` for the two cases, or precalculate the new showing value in a let binding, and then swap it in with an `assoc-in`.

If you can avoid a conditional entirely, even better. I haven't checked it completely, but you might be able to simplify this a little bit further to just do the `update-in`:

```clj
=> (update-in {} [123] not)
{123 true}
```

to use something like this:

```clj
(swap! trace-detail-expansions update-in [:overrides id] not)
```